### PR TITLE
Support various s3 configuration resource types

### PR DESCRIPTION
**Pull request by erran-r7**
In #136 I added convection support for `CorsConfiguration` S3 bucket property. This PR takes it one step further by adding all other configuration resource types to convection.

* [Lifecycle Configuration](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-lifecycleconfig.html)
* [Logging Configuration](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-loggingconfig.html)
* [Notification Configuration](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-notificationconfig.html)
* [Replication Configuration](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-replicationconfiguration.html)
* [Versioning Configuration](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-versioningconfig.html)
* [Website Configuration](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-website.html)

### CODE REVIEW

File: lib/convection/model/template/resource_property/aws_s3_replication_configuration.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+module Convection
+  module Model
+    class Template
+      class ResourceProperty
+        # Represents an {http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-replicationconfiguration.html
+        # Amazon S3 Replication Configuration}
+        class S3ReplicationConfiguration < ResourceProperty
+          property :role, 'Role'
+          property :rules, 'Rules'
```
Comment (erran-r7) [inline lib/convection/model/template/resource_property/aws_s3_replication_configuration.rb]: This should be declared as a list or the `rules` method will not respond to `<<`.

**Comment by erran-r7** [comment]
@athompson-r7 On an internal issue I noticed you mentioning:

> In order to support **xxx service** we need to add support from the [S3 Lambda Notification type](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-bucket-notificationconfig-lambdaconfig.html).

I did that in this PR. Specifically in https://github.com/rapid7/convection/pull/137/commits/da78d738aafc65e27e9efb2aa354f0e6071f7888. 😄

**Comment by erran-r7** [comment]
@abegley-r7 and I had a discussion this morning around adding these new resource types to the DSL. He made me realise that this conversion and the #136 break backwards compatibility (we used to be able to pass in a Hash to define the properties). I'll go ahead and add the backwards compatibility layer before we make any other movements on this PR.

**Comment by bcall-r7** [comment]
Lets close and reopen this against master.

**Comment by erran-r7** [comment]
> Lets close and reopen this against master.

@bcall-r7 👍 will leave open until I have a chance to rebase and update for backwards compatibility. For now I want this visible (also marked as **do not merge**).

**Comment by bturner-r7** [comment]
Actually, I think we should _delete_ the v0.2 branch so we ensure nothing accidentally gets pushed there, and to make sure we don't have something pointed at v0.2 (like Travis or Jenkins, etc.)  Closing that branch will close this PR.

**Comment by erran-r7** [comment]
Go ahead in that case, maybe that's why I couldn't delete the branch earlier.

Sent from my iPhone

On 5 Jul 2016, at 19:06, Brandon Turner <notifications@github.com<mailto:notifications@github.com>> wrote:


Actually, I think we should delete the v0.2 branch so we ensure nothing accidentally gets pushed there, and to make sure we don't have something pointed at v0.2 (like Travis or Jenkins, etc.) Closing that branch will close this PR.

-

**Comment by erran-r7** [comment]
Leaving the branch open to cherry-pick from later. We can delete it after I push up the new PR based-off of master.

**Pull request closed without merging**